docs/advanced.rst

```diff
@@ -505,3 +505,180 @@ Besides the stage directory, additional mount points can be specified through th
 
 
 For a complete list of the available attributes of a specific container platform, the reader is referred to `reference guide <reference.html#container-platforms>`__.
+
+
+Using dependencies in your tests
+--------------------------------
+
+.. versionadded:: 2.21
+
+A ReFrame test may define dependencies to other tests.
+An example scenario is to test different runtime configurations of a benchmark that you need to compile, or run a scaling analysis of a code.
+In such cases, you don't want to rebuild your test for each runtime configuration.
+You could have a build test, which all runtime tests would depend on.
+This is the approach the approach we take with the following example, that fetches, builds and runs several `OSU benchmarks <http://mvapich.cse.ohio-state.edu/benchmarks/>`__.
+We first a create a basic compile-only test, that fetches the benchmarks and builds them for the different programming environments:
+
+.. literalinclude:: ../tutorial/advanced/osu/osu_benchmarks.py
+   :lines: 87-101
+
+There is nothing particular to that test, except perhaps that you can set :attr:`sourcesdir` to ``None`` even for a test that needs to compile something.
+In such a case, you should at least provide the commands that fetch the code inside the :attr:`prebuild_cmd` attribute.
+
+For the next test we need to use the OSU benchmark binaries that we just built, so as to run the MPI ping-pong benchmark.
```

Comment:
```suggestion
For the next test we need to use the OSU benchmark binaries that we just built, in order to run the MPI ping-pong benchmark.
```

Reply:
I don't want to always use "in order to", so I will leave it as it is.